driver/driver.go

```diff
@@ -324,3 +359,19 @@ func SetCustomTLSConfig(dsn string, caPem []byte, certPem []byte, keyPem []byte,
 
 	return nil
 }
+
+// SetDsnOptions sets custom options to the driver that allows modifications to the connection.
+// It requires a full import of the driver (not by side-effects only).
+// Example of supplying a custom option:
+//
+//	driver.SetDsnOptions(map[string]DriverOption{
+//			"my_option": func(c *client.Conn, value string) error {
+//				c.SetCapability(mysql.CLIENT_CAN_HANDLE_EXPIRED_PASSWORDS)
+//				return nil
+//			},
+//		})
+func SetDsnOptions(customOptions map[string]DriverOption) {
```

Comment:
I recommend name it SetDSNOptions, as the golang's style

https://google.github.io/styleguide/go/decisions.html#initialisms

Don't forget to update README's reference to this function name

Reply:
done